client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala

```diff
@@ -1030,23 +1030,31 @@ class LifecycleManager(appId: String, val conf: RssConf) extends RpcEndpoint wit
     slots
   }
 
-  def destroyBuffersWithRetry(applicationId: String, shuffleId: Int,
-    worker: WorkerResource): (util.List[String], util.List[String]) = {
+  def destroyBuffersWithRetry(
+      applicationId: String,
+      shuffleId: Int,
+      worker: WorkerResource): (util.List[String], util.List[String]) = {
     val failedMasters = new util.LinkedList[String]()
     val failedSlaves = new util.LinkedList[String]()
 
     val shuffleKey = Utils.makeShuffleKey(applicationId, shuffleId)
-    worker.asScala.foreach(entry => {
-      var res = requestDestroy(entry._1.endpoint,
-        Destroy(shuffleKey, entry._2._1.asScala.map(_.getUniqueId).asJava,
-          entry._2._2.asScala.map(_.getUniqueId).asJava))
-      if (res.status != StatusCode.Success) {
-        res = requestDestroy(entry._1.endpoint,
-          Destroy(shuffleKey, res.failedMasters, res.failedSlaves))
+    worker.asScala.foreach { case (workerInfo, (masterLocation, slaveLocation)) =>
+      val destroy = Destroy(shuffleKey,
+        masterLocation.asScala.map(_.getUniqueId).asJava,
+        slaveLocation.asScala.map(_.getUniqueId).asJava)
+      var res = requestDestroy(workerInfo.endpoint, destroy)
+      res.status match {
```

Comment:
Since we only handle un-success cases here, I think if(!success) is more clear.

Reply:
DOne